include/geometric_shapes/bodies.h

```diff
@@ -384,12 +384,10 @@ class Box : public Body
 
   // pose/padding/scaling-dependent values & values computed for convenience and fast upcoming computations
   Eigen::Vector3d center_;
-  Eigen::Vector3d normalL_;
-  Eigen::Vector3d normalW_;
-  Eigen::Vector3d normalH_;
+  Eigen::Matrix3d invRot_;
 
-  Eigen::Vector3d corner1_;
-  Eigen::Vector3d corner2_;
+  Eigen::Vector3d corner1_;  //!< The translated, but not rotated min corner
```

Comment:
Okay, I can rename them for noetic. As I'd still like to see this in melodic, should I split this PR into two and try to make this one compatible with melodic? In my view, this approach would be easier than first merging to noetic and then backporting...

As discussed earlier, the API/ABI breaks can be avoided at the cost of minor uglification of code.

Reply:
To maintain ABI you would need to keep the axis vectors instead of `invRot_`. More ugly, but ok.
The other addition I have seen is `std::map<unsigned int, unsigned int> triangle_for_plane_`.
Didn't look for what it is actually used. Replacing this by a global variable, I would actually *not* support. Why is it needed now? Just to be faster?

Reply:
Walking on a thin ice, I guess a `Matrix3d` could take up the same amount of memory as three `Vector3d`s. But that could run in many issues once memory alignment comes into play. So I rather give up this idea.

`invRot` would have to be recomputed every time it is used, but as it is only a transpose, it shouldn't be that bad in melodic. I can also imagine storing the rows/cols of the inverted matrix in the abandoned vectors, but I bet the performance could be even worse than transposing it where needed.

The `triangle_for_plane` is really required for efficiency - otherwise mesh-ray intersection tests would need to re-enumrate over all triangles to find which plane the current one belongs to. The global variable trick is something Gazebo people are willing to accept if it is just "temporary" to get a feature to a released version; the development version then implements the nicer, but ABI-breaking version. An example is here: https://github.com/osrf/gazebo/commit/415be4050e955129956084bcb7bb4c2b436cd943 . It creates a global map, which stores references to created objects (ConvexMesh in this case) and maps them to the structure that is needed to be added. Ugly, but works.

Actually, I'd consider making the whole `MeshData` PIMPL for noetic (i.e. moving it to the .cpp file), so that storing additional data will not cause an ABI break any more. The same could be done for the other bodies, too, but there would be the additional cost of dynamic memory allocation. The ConvexMesh allocates memory anyways, so adding one more unique_ptr should not be a problem.

Reply:
I implemented ABI compatibility using the global variable.